### ISSUE TITLE
ServiceAccount name not propagating to Deployment

### ISSUE THREAD

**Issue by bcbrockway** [kubernetes-sigs/kustomize#1480]
I've found an issue where if you import a Deployment from a base, `1-shared-base`, add a ServiceAccount and a patch to add that service account to the Deployment in an overlay, `2-app-overlay`, and then pull that into another overlay, `3-dev-overlay`, the `serviceAccountName` in the Deployment no longer gets updated.

If you run `kustomize build .` in `2-app-overlay` however, the field updates fine.

I've created a repo to demonstrate the issue [here](https://github.com/bcbrockway/kustomize-serviceaccount-issue).

**Comment by jbrette** [comment]
@bcbrockway This is fixed by the following [PR](https://github.com/kubernetes-sigs/kustomize/pull/1446)

```bash
kustomize build 3-dev-overlay/
```

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: dev-myapp-app
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: dev-myapp-app
spec:
  replicas: 2
  strategy:
    rollingUpdate:
      maxSurge: 15%
      maxUnavailable: 10%
    type: RollingUpdate
  template:
    metadata:
      labels:
        logformat: json
        tier: frontend
    spec:
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - podAffinityTerm:
              topologyKey: kubernetes.io/hostname
            weight: 100
      containers:
      - args: []
        command: []
        image: container-image
        livenessProbe:
          httpGet:
            path: /healthz
            port: http
          initialDelaySeconds: 90
        name: main
        ports:
        - containerPort: 8000
          name: http
        readinessProbe:
          httpGet:
            path: /readiness
            port: http
          initialDelaySeconds: 30
        resources:
          limits:
            cpu: 200m
            memory: 256Mi
          requests:
            cpu: 100m
            memory: 128Mi
      serviceAccountName: dev-myapp-app
```

**Comment by jbrette** [comment]
@bcbrockway Can you check that your issue in now fixed since #1446 has been merged.

**Comment by bcbrockway** [comment]
@jbrette, cloned and built from source and it worked as expected, thanks!
